kiosk/src/Components/AddingGame.tsx

```diff
@@ -0,0 +1,161 @@
+import { useEffect, useState } from "react";
+import { Kiosk } from "../Models/Kiosk";
+import { KioskState } from "../Models/KioskState";
+import configData from "../config.json"
+import "../Kiosk.css";
+import AddGameButton from "./AddGameButton";
+import {QRCodeSVG} from 'qrcode.react';
+interface IProps {
+    kiosk: Kiosk
+  }
+
+const AddingGame: React.FC<IProps> = ({ kiosk }) => {
+    // TODO: update the urls to be more flexible for production code.
+    const [kioskCode, setKioskCode] = useState("");
+    const [renderQRCode, setRenderQRCode] = useState(true);
+    const [menuButtonSelected, setMenuButtonState] = useState(false);
+    const [qrCodeButtonSelected, setQrButtonState] = useState(false);
+    const kioskCodeUrl = "http://localhost:3000/static/kiosk/";
+
+    const updateLoop = () => {
+        if (!menuButtonSelected && kiosk.gamepadManager.isDownPressed()) {
+            setMenuButtonState(true);
+            if (qrCodeButtonSelected) {
+                setQrButtonState(false);
+            }
+        }
+        if (menuButtonSelected && kiosk.gamepadManager.isAButtonPressed()) {
+            kiosk.showMainMenu();
+        }
+        if (!renderQRCode && kiosk.gamepadManager.isUpPressed()) {
+            setMenuButtonState(false);
+            setQrButtonState(true);
+        }
+        if (qrCodeButtonSelected && kiosk.gamepadManager.isAButtonPressed()) {
+            setRenderQRCode(true);
+        }
+    }
+
+    useEffect(() => {
+        let intervalId: any = null;
+        intervalId = setInterval(() => {
+            updateLoop();
+        }, configData.GamepadPollLoopMilli);
+        
+        return () => {
+            if (intervalId) {
+                clearInterval(intervalId);
+            }
+        };
+
+    });
+
+    useEffect(() => {
+        async function generateKioskCode() {
+            const codeGenerationUrl = "https://staging.pxt.io/api/kiosk/newcode"
```

Comment:
we've deployed backend semi recently so should be fine using `https://makecode.com/api/kiosk/newcode` / etc I believe?

Reply:
Oh, good to know! I'll test it out.